zcash_client_sqlite/src/wallet/init/migrations/transparent_gap_limit_handling.rs

```diff
@@ -708,18 +739,30 @@ impl<P: consensus::Parameters, C: Clock, R: RngCore> RusqliteMigration for Migra
         // so we can regenerate the gap limit worth of addresses for each account that we
         // recorded.
         #[cfg(feature = "transparent-inputs")]
-        for account_id in account_ids {
+        for (account_id, (uivk, ufvk)) in account_ids {
             for key_scope in [KeyScope::EXTERNAL, KeyScope::INTERNAL] {
                 use ReceiverRequirement::*;
-                generate_gap_addresses(
-                    conn,
-                    &self.params,
-                    AccountRef(account_id),
-                    key_scope,
-                    &GapLimits::default(),
-                    UnifiedAddressRequest::unsafe_custom(Allow, Allow, Require),
-                    false,
-                )?;
+                let gap_limit = match key_scope {
```

Comment:
Add a comment that the code below duplicates `generate_gap_addresses` (and say why it does so; I believe it is to avoid the `get_account_internal` call).